### PR TITLE
Feature: modularized functions

### PR DESCRIPTION
**Pull request by wovalle**
My intention with this PR is to give you an example in how we can make testable functions by injecting its dependencies.

Now we only need to write a couple of mocks and stubbing a couple of functions in order to unit-test purgePlayers.

This is not the final form of the project by any means, but is a good start.

Notes:
- I didn't touch `purgeParties` function because is likely to die on its own.
- Must be merged after #2

### CODE REVIEW

File: src/db.ts [hunk 1]
```diff
@@ -3,13 +3,64 @@ import * as admin from 'firebase-admin';
...
+  getPartyById(id) {
+    return this.db
+      .collection('parties')
+      .doc(id)
+      .get()
+      .then(ref => ref.data())
+      .catch(() => null);
+  }
+
+  killPlayerInParty(playerId, partyId) {
```
Comment (vetom) [inline src/db.ts]: I think `killPlayerInGame` is a more accurate name.

**Review by vetom — APPROVED** [review]

File: src/db.ts [hunk 2]
```diff
@@ -3,13 +3,64 @@ import * as admin from 'firebase-admin';
 let instance = null;
 
 class Db {
-  db: FirebaseFirestore.Firestore;
+  private db: FirebaseFirestore.Firestore;
```
Comment (mamodom) [inline src/db.ts:6]: :+1:

File: src/functions/purgePlayers.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    const { [player.uid]: playerToRemove, ...remainingPlayers } = party.players;
+
+    const updatedParty = {
+      ...party,
+      players: remainingPlayers,
+    };
+
+    logger.log(`Info: player ${player.uid} will be removed from party`);
+    await db.setParty(player.partyId, updatedParty);
```
Comment (mamodom) [inline src/functions/purgePlayers.ts]: You can perform an update (see [here](https://github.com/lycan-city/werewolf-virtual-cards/blob/d4428c11190cf06e4e17713bd8543ee8bc6f206e/src/db.js#L131))

File: src/db.ts [hunk 4]
```diff
@@ -3,13 +3,64 @@ import * as admin from 'firebase-admin';
...
+      .collection('parties')
+      .doc(partyId)
+      .set(data);
+  }
+
+  updatePlayersKeepAlive(partyId, players) {
+    return this.db
+      .collection('keepAlive')
+      .doc(partyId)
+      .set(players);
```
Comment (mamodom) [inline src/db.ts:51]: You can perform an update

**Pull request merged**